app/models/calculation_result.rb

```diff
@@ -62,143 +66,205 @@ def assessed_capital
+  def client_owns_additional_property?
+    capital_items(:properties)[:additional_properties].present?
   end
 
-  def additional_property_value
-    monetise(additional_property.first[:value])
+  def partner_owns_additional_property?
+    capital_items(:properties, "partner_")&.dig(:additional_properties).present?
   end
 
-  def additional_property_mortgage
-    monetise(additional_property.first[:outstanding_mortgage])
+  def client_additional_property_rows
+    additional_property_rows
   end
 
-  def additional_property_equity
-    monetise(additional_property.first[:assessed_equity])
+  def partner_additional_property_rows
+    additional_property_rows(prefix: "partner_")
   end
 
   def vehicle_owned?
     capital_items(:vehicles).any?
   end
 
-  def vehicle_value
-    monetise(capital_items(:vehicles).sum(0) { |z| z.fetch(:value) })
-  end
-
-  def vehicle_outstanding_payments
-    monetise(capital_items(:vehicles).sum(0) { |z| z.fetch(:loan_amount_outstanding) })
+  def partner_vehicle_owned?
+    capital_items(:vehicles, "partner_")&.any?
   end
 
-  def vehicle_disregards
-    value = capital_items(:vehicles).sum(0) { |z| z.fetch(:value) }
-    pcp = capital_items(:vehicles).sum(0) { |z| z.fetch(:loan_amount_outstanding) }
-    assessed_value = capital_items(:vehicles).sum(0) { |z| z.fetch(:assessed_value) }
-    monetise(value - pcp - assessed_value)
+  def client_vehicle_rows
+    vehicle_rows
   end
 
-  def vehicle_assessed_value
-    monetise(capital_items(:vehicles).sum(0) { |z| z.fetch(:assessed_value) })
+  def partner_vehicle_rows
+    vehicle_rows(prefix: "partner_")
   end
 
   def client_capital_rows
-    data = {
-      # unfortunately CFE returns capital items as strings rather than numbers for some reason.
-      second_property: capital_items(:properties).fetch(:additional_properties).sum(0) { |p| p.fetch(:net_equity).to_i },
-      savings: capital_items(:liquid).sum(0) { |z| z.fetch(:value).to_i },
-    }
-    data.transform_values { |value| monetise(value) }
+    capital_rows
   end
 
   def client_capital_subtotal_rows
-    data = {
-      total_capital: api_response.dig(:result_summary, :capital, :total_capital),
-      total_capital_limit: api_response.dig(:result_summary, :capital, :proceeding_types).map { |pt| pt.fetch(:upper_threshold) }.min,
-      pensioner_capital_disregard: api_response.dig(:result_summary, :capital, :pensioner_capital_disregard),
-      smod_disregard: api_response.dig(:result_summary, :capital, :subject_matter_of_dispute_disregard),
-    }
+    capital_subtotal_rows
+  end
 
-    data.transform_values { |value| monetise(value) }
+  def partner_capital_rows
+    capital_rows(prefix: "partner_")
+  end
+
+  def partner_capital_subtotal_rows
+    capital_subtotal_rows(prefix: "partner_")
   end
 
 private
 
   attr_reader :api_response
 
-  def capital_items(key)
-    api_response.dig(:assessment, :capital, :capital_items, key)
+  def capital_items(key, prefix = nil)
```

Comment:
I think I would prefer '' over nil here to show that the prefix is empty rather than missing. WDYT?

Reply:
Sounds sensible, will do.